### PR TITLE
[test] Modernise IOR checks

### PR DESCRIPTION
**Pull request by victorusu**

### CODE REVIEW

**Comment by victorusu** [comment]
@jenkins-cscs retry all

**Review by teojgo — APPROVED** [review]

File: cscs-checks/system/io/ior_check.py [hunk 1]
```diff
@@ -12,16 +12,27 @@
...
-        self.prerun_cmds = ['mkdir -p ' + self.test_dir]
-        self.test_file = os.path.join(self.test_dir, 'ior')
+    base_dir = parameter(['/scratch/e1000',
+                          '/scratch/snx3000tds',
+                          '/scratch/snx3000',
+                          '/scratch/shared/fulen',
+                          '/users'])
+    username = getpass.getuser()
+    time_limit = '5m'
+
```
Comment (vkarak) [inline cscs-checks/system/io/ior_check.py:22]: No newline here.

File: cscs-checks/system/io/ior_check.py [hunk 2]
```diff
@@ -12,16 +12,27 @@
...
+    maintainers = ['SO', 'GLR']
+    tags = {'ops', 'production', 'external-resources'}
+
+    @run_after('init')
+    def set_description(self):
+        self.descr = f'IOR check ({self.base_dir})'
+
+    @run_after('init')
+    def extend_tags_based_on_fs(self):
```
Comment (vkarak) [inline cscs-checks/system/io/ior_check.py:31]:
```suggestion
    def add_fs_tags(self):
```

File: cscs-checks/system/io/ior_check.py [hunk 3]
```diff
@@ -76,86 +87,88 @@ def __init__(self, base_dir):
...
         self.sourcesdir = os.path.join(self.current_system.resourcesdir, 'IOR')
-        self.executable = os.path.join('src', 'C', 'IOR')
-        self.build_system = 'Make'
-
-        vpe = 'valid_prog_environs'
-        penv = self.fs[base_dir][cur_sys].get(vpe, ['builtin'])
-        self.valid_prog_environs = penv
 
+    @run_after('init')
+    def set_build_systems(self):
```
Comment (vkarak) [inline cscs-checks/system/io/ior_check.py:116]:
```suggestion
    @run_before('compile')
    def prepare_build(self):
```

If you prefer to have everything as a post-init hook, I'm fine.

File: cscs-checks/system/io/ior_check.py [hunk 4]
```diff
@@ -76,86 +87,88 @@ def __init__(self, base_dir):
...
-        self.maintainers = ['SO', 'GLR']
-
-        systems_to_test = ['dom', 'daint', 'eiger', 'pilatus']
-        if self.current_system.name in systems_to_test:
-            self.tags |= {'production', 'external-resources'}
+        self.test_dir = os.path.join(self.base_dir, self.username, '.ior')
+        self.prerun_cmds = ['mkdir -p ' + self.test_dir]
 
     @run_before('run')
     def set_exec_opts(self):
```
Comment (vkarak) [inline cscs-checks/system/io/ior_check.py:131]: I would combine this with the above hook in something called `prepare_run`.

File: cscs-checks/system/io/ior_check.py [hunk 5]
```diff
@@ -76,86 +87,88 @@ def __init__(self, base_dir):
...
-            '*': self.fs[base_dir]['reference']
-        }
-
-        self.maintainers = ['SO', 'GLR']
-
-        systems_to_test = ['dom', 'daint', 'eiger', 'pilatus']
-        if self.current_system.name in systems_to_test:
-            self.tags |= {'production', 'external-resources'}
+        self.test_dir = os.path.join(self.base_dir, self.username, '.ior')
+        self.prerun_cmds = ['mkdir -p ' + self.test_dir]
```
Comment (vkarak) [inline cscs-checks/system/io/ior_check.py:128]:
```suggestion
        self.prerun_cmds = [f'mkdir -p {self.test_dir}']
```

File: cscs-checks/system/io/ior_check.py [hunk 6]
```diff
@@ -76,86 +87,88 @@ def __init__(self, base_dir):
...
+@rfm.simple_test
 class IorWriteCheck(IorCheck):
-    def __init__(self, base_dir):
-        super().__init__(base_dir)
-        self.executable_opts += ['-w', '-k']
+    executable_opts += ['-w', '-k']
+    tags |= {'write'}
+
+    @run_after('init')
+    def set_sanity_perf_patterns(self):
```
Comment (vkarak) [inline cscs-checks/system/io/ior_check.py:150]: I would split that and use the `@sanity_function` syntax for the sanity patterns.

File: cscs-checks/system/io/ior_check.py [hunk 7]
```diff
@@ -76,86 +87,88 @@ def __init__(self, base_dir):
...
-                        ['/scratch/shared/fulen'])
 class IorReadCheck(IorCheck):
-    def __init__(self, base_dir):
-        super().__init__(base_dir)
-        self.executable_opts += ['-r']
+    executable_opts += ['-r']
+    tags |= {'read'}
+
+    @run_after('init')
+    def set_sanity_perf_patterns(self):
```
Comment (vkarak) [inline cscs-checks/system/io/ior_check.py:164]: See my comment above.

**Review by vkarak — CHANGES_REQUESTED** [review]

**Comment by vkarak** [comment]
I am fixing this PR

**Review by vkarak — APPROVED** [review]

**Pull request merged**